lib/job.js

```diff
@@ -131,6 +131,28 @@ class Job extends BaseModel {
 
         return factory.list(listConfig);
     }
+
+    /**
+     * Remove all builds associated with this job and the job itself
+     * @return {Promise}        Resolves to null if remove successfully
+     */
+    remove() {
+        const removeBuilds = (() =>
+            // Iterate through the builds and remove them
+            this.getBuilds().then(builds => {
+                if (builds.length > 0) {
+                    builds.forEach(build => Promise.all(build.remove()));
```

Comment:
Oh... I'm not sure how promises will work with the way you wrote this function.  I can tell you the `Promise.all` code concerns me.